File: lang/kudu/test/bench.js
class Point {
   var _x : Number = 0
   var _y : Number = 0
   function move(x,y) {
      this._x = x
      this._y = y
   }
}

class Point3D extends Point {
   var _z : Number = 0
   function move(x, y, z) {
      super.move(x, y)
      this._z = z
   }
}

var p = new Point3D()
p._z = 42
print(p._z)

for i=1, 1e7 {
   p.move(i, i + 1, i + 2)
}
print(p._x)
function cheese(mesg) {
    print("mesg:", mesg)
}
cheese("Hey!")

File: lang/kudu/test/bench.pl
package Point;

use Moose;

has x => ( isa => 'Num', is => 'rw', default => 0 );
has y => ( isa => 'Num', is => 'rw', default => 0 );

sub move {
    my ($self, $x, $y) = @_;
    $self->x($x);
    $self->y($y);
}

__PACKAGE__->meta->make_immutable;

package Point3D;

use Moose;
extends qw/Point/;

has z => ( isa => 'Num', is => 'rw', default => 0 );

sub move {
    my ( $self, $x, $y, $z ) = @_;
    $self->SUPER::move($x, $y);
    $self->z($z);
}

__PACKAGE__->meta->make_immutable;

package main;

my $p = Point3D->new();
for ( 1 .. 10_000_000 ) {
    $p->move($_, $_ + 1, $_ + 2);
}

1;
